Source/cryptalgo/Hash.h

```diff
@@ -115,17 +131,33 @@ namespace Crypto {
         static const EnumHashType Type = HASH_SHA1;
         static const uint8_t Length = 20;
 
+        static uint32_t ContextLength()
+        {
+            return static_cast<uint32_t>(sizeof(_context));
+        }
+
+        inline const uint8_t* CurrentContext() const
```

Comment:
Why not return here the inline const Context& CurrentContex() const. Leave it up to the receiver on how to interpret and store this.

Reply:
done